processor/src/main/java/org/mapstruct/ap/internal/conversion/JavaStreamToCollectionConversion.java

```diff
@@ -41,7 +41,9 @@
 
     @Override
     protected String getToExpression(ConversionContext conversionContext) {
-        return "<SOURCE>.collect( Collectors." + collector( conversionContext ) + " )";
+        return "Optional.ofNullable( <SOURCE> )" +
```

Comment:
ah.. we should fix that someday. I'll try to take that on when I do the refactoring on the templates for the property mapping. For now.. leave it.